pkg/providers/v1/aws.go

```diff
@@ -857,6 +857,10 @@ func (c *Cloud) NodeAddressesByProviderID(ctx context.Context, providerID string
 		return nil, err
 	}
 
+	return c.getInstanceNodeAddress(instance)
```

Comment:
Do we actually need the above method anymore? Some of these methods may actually be obsolete after migrating to the Instances V2 API. I would confirm.